### PR TITLE
Feature/performance metrics

### PR DESCRIPTION
**Pull request by rnugent3**

### CODE REVIEW

**Comment by rnugent3** [comment]
@HenryGeorgist would you mind taking a look at my latest commit? I am hoping the structure makes sense for what we need - a results object that holds both performance results and ead results.

File: fda-model/metrics/IContainResults.cs [hunk 1]
```diff
@@ -3,12 +3,9 @@ namespace metrics
...
     public interface IContainResults
     {
-
-        void AddEADEstimate(double eadEstimate, string category);
-
-        double MeanEAD(string category);
-
-        double EADExceededWithProbabilityQ(string category, double exceedanceProbability);
+    
+        public Thresholds thresholds;
```
Comment (HenryGeorgist) [inline fda-model/metrics/IContainResults.cs:7]: I think your property needs to follow capitilization rules

File: fda-model/metrics/IContainResults.cs [hunk 2]
```diff
@@ -3,12 +3,9 @@ namespace metrics
 {
     public interface IContainResults
```
Comment (HenryGeorgist) [inline fda-model/metrics/IContainResults.cs:4]: forcing properties is good. another alternative is to force the signature for your retrieval - how do you expect people to gather data? getting the two objects you provide via properties and drilling down works, an alternative could be something like a method like double MeanEADByCategory(string category); but that might be too specific for our current situation.

**Comment by rnugent3** [comment]
the test I am running - drawn from SimulationShould - appears not to use a stage frequency. I think I am missing something, and would like to discuss. I hope we are close.

**Comment by rnugent3** [comment]
in bootstrap to paired data, the probabilities were exceedance probabilities but the yvals were being computed as if they are non-exceedance probabilities. I changed the probs to non-exceedance probabilities, this way the probabilities and flows or stages are both increasing and the check for monotonicity works with all relationships the same way. However, as expected, this broke some stuff which is to be fixed next.

**Comment by rnugent3** [comment]
I am not sure why the build is failing on GitHub, the software builds successfully on my machine and all tests are passing. 

I am tracking two remaining issues: (1) the compute AEP with iterations test passes with 1 iteration but not with many. The AEP is much different with many iterations. (2) we'll need a cnep unit test.

**Review by HenryGeorgist — APPROVED** [review]

**Pull request merged**